services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -27,92 +48,161 @@
+    otelMetricDimensions = Attributes.builder()
+        .put(getDimensionName(VENICE_STORE_NAME), storeName)
+        .put(getDimensionName(VENICE_REQUEST_METHOD), requestType.getRequestTypeName())
+        .put(getDimensionName(VENICE_CLUSTER_NAME), clusterName)
+        .build();
+
     this.systemStoreName = VeniceSystemStoreUtils.extractSystemStoreType(storeName);
     Rate requestRate = new OccurrenceRate();
     Rate healthyRequestRate = new OccurrenceRate();
     Rate tardyRequestRate = new OccurrenceRate();
-    requestSensor = registerSensor("request", new Count(), requestRate);
+
+    incomingRequestSensor = registerSensor("request", new Count(), requestRate);
+    incomingRequestSensorOtel = metricsRepository.getOpenTelemetryMetricsRepository()
+        .getCounter("incoming_call_count", "Number", "Count of all incoming requests");
+
     healthySensor = registerSensor("healthy_request", new Count(), healthyRequestRate);
     unhealthySensor = registerSensor("unhealthy_request", new Count());
-    unavailableReplicaStreamingRequestSensor = registerSensor("unavailable_replica_streaming_request", new Count());
     tardySensor = registerSensor("tardy_request", new Count(), tardyRequestRate);
+    throttleSensor = registerSensor("throttled_request", new Count());
     healthyRequestRateSensor =
         registerSensor(new TehutiUtils.SimpleRatioStat(healthyRequestRate, requestRate, "healthy_request_ratio"));
     tardyRequestRatioSensor =
         registerSensor(new TehutiUtils.SimpleRatioStat(tardyRequestRate, requestRate, "tardy_request_ratio"));
-    throttleSensor = registerSensor("throttled_request", new Count());
-    errorRetryCountSensor = registerSensor("error_retry", new Count());
     badRequestSensor = registerSensor("bad_request", new Count());
-    badRequestKeyCountSensor = registerSensor("bad_request_key_count", new OccurrenceRate(), new Avg(), new Max());
+    requestSensorOtel = metricsRepository.getOpenTelemetryMetricsRepository()
+        .getCounter("call_count", "Number", "Count of all requests with response details");
+
+    errorRetryCountSensor = registerSensor("error_retry", new Count());
+    retryTriggeredSensorOtel = metricsRepository.getOpenTelemetryMetricsRepository()
+        .getCounter("retry_call_count", "Number", "Count of retries triggered");
+    allowedRetryRequestSensor = registerSensor("allowed_retry_request_count", new OccurrenceRate());
+    allowedRetryRequestSensorOtel = metricsRepository.getOpenTelemetryMetricsRepository()
+        .getCounter("allowed_retry_call_count", "Number", "Count of allowed retry requests");
+    disallowedRetryRequestSensor = registerSensor("disallowed_retry_request_count", new OccurrenceRate());
+    disallowedRetryRequestSensorOtel = metricsRepository.getOpenTelemetryMetricsRepository()
+        .getCounter("disallowed_retry_call_count", "Number", "Count of disallowed retry requests");
```

Comment:
The boilerplate is pretty heavy here... there are a lot of similar cases. Is it conceivable to have some mapping between Tehuti and OTel, which would allow us to abstract over both more cleanly? for example, can we have some sort of API like this:

```
VeniceMetric registerOccurrenceRate(String metricName, String unitName, String description) {
  return new VeniceMetric(
    registerSensor("disallowed_retry_request_count", new OccurrenceRate()),
    metricsRepository.getOpenTelemetryMetricsRepository()
        .getCounter(metricName, unitName, description));
}
```

And then:

```
public class VeniceMetric {
  private final Sensor tehuti;
  private final LongCounter otel;
  public VeniceMetric(Sensor tehuti, LongCounter otel) {
    this.tehuti = tehuti;
    this.otel = otel;
  }

  void record(long count, Attributes dimensions) {
    this.tehuti.record(count);
    this.otel.add(count, otelMetricDimensions);
  }
}
```

The above is a simplified example of course... There would be a few different variations (probably subclasses of `VeniceMetric`...?), e.g. one for occurrence rates, one for histograms, and possibly a few more. But I don't think there would need to be a TON of them, right? I would think just a handful can probably cover all the useful cases.

There would still need to be a big refactoring that touches all/most metrics, but at least it wouldn't be that much boilerplate...

IDK, WDYT 🤔 ?

Reply:
Yes, I agree the boiler plate is pretty heavy. As this is going to be a temporary phase until we retire tehuti (though it's a long way to go) and will look cleaner after that, especially with `MetricEntity`, I didn't invest time on it yet. We can discuss this further and see if we want to do this, maybe after this PR. WDYT?

Reply:
Hmm that seems concerning, IMHO. It could take a very long time before we manage to fully get rid of Tehuti. And in the meantime, we will keep adding new metrics or modifying existing ones. It will be quite burdensome maintenance-wise to carry all this duplicate code in sync across both stacks.

Is there any concern with the path suggested above...?

Reply:
My concern was that opentelemetry and tehuti metrics are not 1:1. 
1. 1 tehuti metric can have multiple aggregations like OccurrenceRate, min, max, etc
2. 1 opentelemetry metric can cover multiple tehuti metrics. 
I will take a stab to structure this together.

Reply:
That is right... but I think the equivalence is roughly between Tehuti's `Sensor` and OTel's "metric".

Tehuti lets us put whatever individual `MeasurableStat` within a given `Sensor`, whereas in OTel the choices are more standardized/constrained (e.g. you get min/max/count/percentiles/whatever as some one-size-fits-all package, whereas in Tehuti you would have needed to specify each of these "stats" explicitly). But although there is a bit of a mismatch here, it doesn't seem insurmountable to me...

I'm envisioning that as part of transitioning to OTel, maybe we'll end up making an existing Tehuti Sensor carry some extra stats it didn't have before but I think that is probably not a big problem. Essentially, I am saying we could coerce our own usage of Tehuti into offering only the same "packages" that OTel offers. And in doing so, we'll achieve a 1:1 mapping between them, and that in turn allows us to have a much cleaner abstraction with little to no boilerplate.